tests/test_association_task.py

```diff
@@ -75,13 +95,40 @@ def test_run(self):
                 [0]):
             self.assertEqual(test_obj_id, expected_obj_id)
 
+    def test_run_trailed_sources(self):
+        """Test the full task by associating a set of diaSources to
+        existing diaObjects when trailed sources are filtered
+        # """
+
+        config = AssociationTask.ConfigClass()
+        config.doLongTrailFilter = True
+        assocTask = AssociationTask(config=config)
+
+        results = assocTask.run(self.diaSources, self.diaObjects, self.exposure)
+
+        print(results.nUpdatedDiaObjects, "results nupdated")
+        print(len(self.diaObjects) - 3, " length of dia objects")
+        self.assertEqual(results.nUpdatedDiaObjects, len(self.diaObjects) - 3)
+        self.assertEqual(results.nUnassociatedDiaObjects, 3)
+        self.assertEqual(len(results.matchedDiaSources),
+                         len(self.diaObjects) - 3)
+        self.assertEqual(len(results.unAssocDiaSources), 1)
+        for test_obj_id, expected_obj_id in zip(
+                results.matchedDiaSources["diaObjectId"].to_numpy(),
+                [1, 2, 3, 4]):
+            self.assertEqual(test_obj_id, expected_obj_id)
+        for test_obj_id, expected_obj_id in zip(
+                results.unAssocDiaSources["diaObjectId"].to_numpy(),
+                [0]):
+            self.assertEqual(test_obj_id, expected_obj_id)
+
```

Comment:
This test block looks very similar to the one in `test_run`: how many of these values are actually different? How many trailed sources do you expect to have in the test dataset, and does that match the changes here?

I think you'd be better off having a separate set of unittests of the new Task itself, in its own test file, so you can confirm that the new Task's `run` is returning the correct things in its Struct.

Reply:
I've created a new unit test called `test_trailedSourceFilter.py`. I wasn't sure if I should leave testing it out completely from the association tests because filtering out does change the expected number of associated sources and objects. As of right now, we haven't run this on a full dataset so it's hard to say how many sources will be filtered out. I'm hoping Erin can do a test run of this before it's merged onto main so we can see the effects. since ap_verify doesn't have any sources that meet the filter criteria.